src/spring-mongo/src/main/java/net/conology/spring/restjsonpath/mongo/NestedValueTestCompiler.java

```
@@ -105,9 +108,10 @@ private MongoElementMatch compileElementMatch(
             if (next instanceof PropertyFilterNode filterNode) {
                 nodes.next();
                 var testNode = parent.compileTestNode(filterNode);
-                switch (testNode) {
-                    case MongoAllOfSelector allOf -> propertyTests.addAll(allOf.getTests());
-                    case MongoPropertyCondition propertyTest -> propertyTests.add(propertyTest);
+                if (testNode instanceof MongoAllOfSelector allOf) {
```

default handling is missing. we should throw an assertion error here, if we drop type system support